metric_tank/http.go

```diff
@@ -127,10 +128,25 @@ func Get(w http.ResponseWriter, req *http.Request, metaCache *MetaCache, aggSett
 			http.Error(w, "unrecognized consolidation function", http.StatusBadRequest)
 			return
 		}
-		log.Debug("===================================")
 		req := NewReq(id, fromUnix, toUnix, minDataPoints, maxDataPoints, consolidator)
+		reqs = append(reqs, req)
```

Comment:
This is wrong.  reqs is initialized with a size of len(targets).  By appending here, you are leaving len(targets) empty items in reqs.

This then leads to metricdef.GetMetricDefinition() being called with an empty ("") metric id, which in turn leads to `2015/12/21 09:44:45 [metricdef.go:235 GetMetricDefinition()] [E] elasticsearch query failed. 2015-12-21 09:44:45.319853618 +0000 UTC: Error [ElasticsearchIllegalArgumentException[No feature for name [metric_index]]] Status [400] [400]` you have been seeing.